script/domVisitor.ts

```diff
@@ -76,7 +86,32 @@ async function visitPages(
       const appendQueryString =
         queryString ? (archiveUrl ? `?archive=${archiveUrl}&${queryString}` : `?${queryString}`)
                     : archiveUrl ? `?archive=${archiveUrl}` : '';
-      await page.goto(`${rootUrl}${pageUrl}${appendQueryString}`);
+      // validateLinks should persist within the SPA so we allow it when clicking links
+      if (appendQueryString && appendQueryString !== '?validateLinks') {
+        await page.goto(`${rootUrl}${pageUrl}${appendQueryString}`);
+      } else {
+        // If there's no queryString, we can click the content links instead which are much faster
+        // Prefer using env variables instead of queryStrings when possible
+        const pageComponents = pageUrl.split('/');
+        const slug = pageComponents[pageComponents.length - 1];
+        const linkSelector = `a[href="${slug}"]`;
```

Comment:
but i don think the archive url will get written into the anchor, so you may need to loosen up this check, maybe `a[href^="${slug}"]` or whatever the starts with check is